astropy/nddata/tests/test_decorators.py

```diff
@@ -58,6 +58,84 @@ def test_pass_nddata():
     assert unit_out is unit_in
 
 
+@pytest.mark.parametrize(
+    "func",
+    (
+        lambda *, data, wcs, unit=None: (data, wcs, unit),
+        lambda *, wcs=None, data, unit=None: (data, wcs, unit),
+    ),
+)
+def test_pass_nddata_kwarg_only(func):
+    wrapped_function = support_nddata(func)
+
+    data_in = np.array([1, 2, 3])
+    wcs_in = WCS(naxis=1)
+    unit_in = u.Jy
+
+    nddata_in = NDData(data_in, wcs=wcs_in, unit=unit_in)
+
+    data_out, wcs_out, unit_out = wrapped_function(data=nddata_in)
+
+    assert data_out is data_in
+    assert wcs_out is wcs_in
+    assert unit_out is unit_in
+
+
+@pytest.mark.parametrize(
+    "func",
+    (
+        lambda data, wcs=None, mask=None, /, *, unit=None: (data, wcs, unit, mask),
+        lambda data, wcs=None, /, mask=None, *, unit: (data, wcs, unit, mask),
```

Comment:
just for fun, and because parametrization makes adding these almost free
```suggestion
        lambda data, wcs=None, /, mask=None, *, unit: (data, wcs, unit, mask),
        lambda wcs=None, /, data, mask=None, *, unit: (data, wcs, unit, mask),
        lambda wcs=None, /, mask=None, *, data, unit: (data, wcs, unit, mask),
```